connector/apmconnector/trace_connector.go

```diff
@@ -0,0 +1,65 @@
+// Copyright New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package apmconnector // import "apmconnector"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+type ApmTraceConnector struct {
+	config    *Config
+	logger    *zap.Logger
+	sqlparser *SQLParser
+
+	tracesConsumer consumer.Traces
+}
+
+func (c *ApmTraceConnector) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: true}
+}
+
+func (c *ApmTraceConnector) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	MutateSpans(c.logger, c.sqlparser, td)
+	return c.tracesConsumer.ConsumeTraces(ctx, td)
+}
+
+func (c *ApmTraceConnector) Start(_ context.Context, _ component.Host) error {
+	c.logger.Info("Starting the New Relic APM Trace Connector")
+	if c.config.ApdexT == 0 {
+		c.config.ApdexT = 0.5
```

Comment:
I'm not sure if this ApdexT will change, but the default value could be a package level const